### PR TITLE
Fix: use forwardref for nullability

### PR DESCRIPTION
**Pull request by JeehyaeDance**
### Description

This PR is for "[v6] Expand ref type, use React.forwardRef #969". Wrapped `carousel` component with `React.forwardRef` and replaced `innerRef` prop with a native-to-React `ref` prop.

Fixes # 969 (issue)

#### Type of Change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist

- [x] My code follows the style guidelines of this project (I have run `pnpm run lint`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (I have run `pnpm run test:ci-with-server`/`pnpm run test`)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have recorded any user-facing fixes or changes with `pnpm changeset`.
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/nuka/src/hooks/use-forward-ref.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useEffect, useRef, ForwardedRef } from 'react';
+
+export const useForwardRef = <T>(ref: ForwardedRef<T>) => {
```
Comment (JeehyaeDance) [inline packages/nuka/src/hooks/use-forward-ref.ts:3]: `const carouselRef = ref || defaultCarouselRef;`
At first, I tried to pass `ref` that was created from `React.forwardRef` to line 123 in `carousel.tsx`, however, there was a typescript error. (Property 'current' does not exist on type '(instance: HTMLDivElement | null) => void'.ts)
That's why I created this hook to handle null and I referenced it from [here](https://github.com/facebook/react/issues/24722).

File: packages/nuka/src/carousel.tsx [hunk 2]
```diff
@@ -18,672 +18,682 @@ import {
...
-  const slideCount = filteredSlides.length;
-
-  const slidesToScroll =
-    animation === 'fade' ? slidesToShow : propsSlidesToScroll;
-
-  const [currentSlide, setCurrentSlide] = useState<number>(() =>
-    getDefaultSlideIndex(
+import { useForwardRef } from './hooks/use-forward-ref';
+
+// eslint-disable-next-line react/display-name
```
Comment (fritz-c) [inline packages/nuka/src/carousel.tsx:23]: You can avoid having to disable this rule by naming the function argument like `function Carousel(rawProps, ref): React.ReactElement {`

File: packages/nuka/src/hooks/use-forward-ref.ts (hunk 1 above)
Comment (fritz-c) [reply packages/nuka/src/hooks/use-forward-ref.ts:3]: Makes sense. Nice find 👍

**Review by fritz-c — CHANGES_REQUESTED** [review]
You'll need to add a changeset. We might want to look into any other breaking changes for v6 to release at the same time, so maybe the target branch should be a new one called `6.x` or something like that, instead of `main`.

**Comment by fritz-c** [comment]
I only spotted a couple other breaking changes that had been proposed. There weren't any in GitHub issues, but there were a couple `@deprecated`-marked exports in the code that we could get rid of for v6.

**Review by fritz-c — APPROVED** [review]

**Pull request merged**